File: leetcode/217-contains-duplicate/solution.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        final Set<Integer> set = new HashSet<>();

        for (int num : nums) {
            if (!set.add(num)) {
                return true;

            }
        }
        
        return false;

    }
}

File: leetcode/217-contains-duplicate/solution.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
			unordered_set<int> set;

			for (int i = 0; i < nums.size(); i++) {
				int s = set.size();
				set.insert(nums[i]);
				if (s == set.size()) {
					return true;
				}

			}
			return false;
        
    }
};
